### PR TITLE
Proprly logout from OAUTH provider

### PR DESCRIPTION
**Pull request by michaeladda**
See https://auth0.com/docs/logout for more details

### CODE REVIEW

File: passport/src/auth-handler.ts [hunk 1]
```diff
@@ -140,7 +141,18 @@ export function authRouter(): express.Express {
 
   app.get('/logout', (req, res) => {
     req.logout();
-    res.redirect('/');
+    if (isFake(strategy)) {
```
Comment (arielshaqed) [inline passport/src/auth-handler.ts]: I'd rather keep the `isFake` call outside of the callback.

File: passport/src/auth-handler.ts [hunk 2]
```diff
@@ -140,7 +141,18 @@ export function authRouter(): express.Express {
...
+      res.redirect('/');
+    } else {
+      const oauthDomain = process.env.OAUTH_DOMAIN!;
+      const baseUrl = process.env.RESHUFFLE_APPLICATION_DOMAINS!.split(',')[0];
+      const clientId = process.env.OAUTH_CLIENT_ID!;
+      const params = new URLSearchParams({
+        returnTo: `https://${baseUrl}`,
+        client_id: clientId,
+      }).toString();
+      res.redirect(`https://${oauthDomain}/v2/logout?${params}`);
```
Comment (arielshaqed) [inline passport/src/auth-handler.ts]: This also clears the session cookies?  Cool...

Comment (michaeladda) [reply passport/src/auth-handler.ts]: it doesn't clear it. it invalidates it afaict.

**Comment by michaeladda** [comment]
passed in https://ci.binaris.io/job/gobinaris/job/precious/job/master/9623/ so merging

**Pull request merged**